code/API_definitions/consent-info.yaml

```diff
@@ -264,71 +304,95 @@ components:
+        Provides information about the data processing validity status for the requested scope(s) and Purpose. It contains details about the applicable Legal Basis, whether it is currently valid for processing data, and if not, the reason why it is not valid. This information helps API Consumers understand the current state of Consent or other Leal Basis for processing data.
+        It may be required more than one array item for the requested scope(s) and Purpose, e.g. when the requested scopes are related to multiple APIs.
       items:
-        $ref: "#/components/schemas/PrivacyStatusObject"
+        $ref: "#/components/schemas/statusInfoObject"
       minItems: 1
-    PrivacyStatusObject:
+    statusInfoObject:
       type: object
       required:
         - scopes
         - purpose
-        - consentRequired
+        - legalBasis
+        - statusValidForProcessing
       properties:
         scopes:
           $ref: "#/components/schemas/Scopes"
         purpose:
           $ref: "#/components/schemas/Purpose"
-        consentRequired:
+        legalBasis:
+          type: string
+          description: |
+              The applicable Legal Basis for the requested scope(s) and Purpose. It uses terms from the W3C [Data Privacy Vocabulary](https://w3c.github.io/dpv/2.1/) (DPV).
+          enum:
+            - dpv:Consent
+            - dpv:LegitimateInterest
+            - dpv:Contract
+            - dpv:LegalObligation
+            - dpv:PublicInterest
+            - dpv:VitalInterest
+        statusValidForProcessing:
           type: boolean
           description: |
-            * `true` - The Consent is required for the requested scope(s) and Purpose.
-            * `false` - The Consent is not required for the requested scope(s) and Purpose.
-        consentStatus:
+              Boolean flag that shows whether the specified Legal Basis is currently valid for processing data related to the indicated scope(s) and Purpose.
+              * `true` - The specified `legalBasis` is currently valid and permits the requested data processing. For example, necessary user Consent has been obtained; a legitimate interests assessment resulted in a positive outcome, with no overriding objections; or the processing is deemed necessary under an existing contract.
+              * `false` - The specified `legalBasis` is not valid for processing the requested scope(s) and Purpose. The reason for this is indicated in the `statusReason` field. This may be because the User has not yet provided Consent, has opted out of processing based on legitimate interest or because the Legal Basis is no longer applicable for other reasons.
+        statusReason:
           type: string
           enum:
             - PENDING
```

Comment:
Do we need REQUESTED and UNKNOWN instead of pending?
REQUESTED would mean, end user has been asked, but the end user is yet to respond.

UNKNOWN is when the end user has not been asked yet?

Reply:
'PENDING' is defined as follows: The specified legal basis is not yet established or fully validated for the requested data processing. For example, the user has not yet provided consent under the 'dpv:Consent' legal basis.' This is not specific to dpv:Consent, so it has been defined in a generic way to cover all possible statuses for the different legal bases where they are not yet fully ready or established. If you look at the W3C DPV, you will see that it defines many statuses that could potentially map here. I would leave it as it is.

Reply:
@jpengar Yes, I am aware that we don't follow W3C DPV yet for consent status, but I would like to keep it as close as possible so that the transition is easier.